Browser_IDE/executionEnvironment_Internal.js

```diff
@@ -420,7 +420,12 @@ window.addEventListener('message', async function(m){
         if (m.data.type == "rename"){
             FS.rename(m.data.oldPath,m.data.newPath);
         }
-
+        
+        if (m.data.type == "readFile"){
+            let a= FS.readFile(m.data.path);
+            console.log(a);
+            
+        }
```

Comment:
Right now this reads the file and logs it to the console - a good start! To return it, you need to send a message _back_ to the main window. To keep this simple, just delete the `console.log` and paste this in:
```javascript
            parent.postMessage({
                type: "callback",
                callbackID: m.data.callbackID,
                result: a,
                error: undefined,
            }, "*");
            return;

```
Also maybe rename `a` to something like `fileData` 😃